modules/cas_disk/exp_obj.c

```diff
@@ -505,61 +505,54 @@ int casdsk_exp_obj_create(struct casdsk_disk *dsk, const char *dev_name,
 
 	result = casdsk_exp_obj_alloc(dsk);
 	if (result)
-		goto error_alloc;
+		goto error_exp_obj_alloc;
 
 	exp_obj = dsk->exp_obj;
 
 	exp_obj->dev_name = kstrdup(dev_name, GFP_KERNEL);
 	if (!exp_obj->dev_name) {
 		__casdsk_exp_obj_release(exp_obj);
 		result = -ENOMEM;
-		goto error_strdup;
+		goto error_kstrdup;
 	}
 
 	result = _casdsk_exp_obj_init_kobject(dsk);
 	if (result) {
 		__casdsk_exp_obj_release(exp_obj);
-		goto error_kobject;
+		goto error_init_kobject;
 	}
 
 	if (!try_module_get(owner)) {
 		CASDSK_DEBUG_DISK_ERROR(dsk, "Cannot get reference to module");
 		result = -ENAVAIL;
-		goto error_module;
+		goto error_module_get;
```

Comment:
In case of failing to increment module reference count or any further errors `__casdsk_exp_obj_release()` won't be called